src/main/kotlin/kweb/prelude.kt

```diff
@@ -434,16 +437,38 @@ abstract class ValueElement(open val element: Element, val kvarUpdateEvent: Stri
             _valueKvar = v
         }
 
+
     /**
      * Automatically update `toBind` with the value of this INPUT element when `updateOn` event occurs.
      */
+
+    //I wasn't sure where this data class and function should go
+    @Serializable
+    data class DiffData(val prefixEnd: Int, val postFixOffset: Int, val diff: String)
+
+    private fun applyDiff(oldString: String, diffData: DiffData) : String {
+        return if (diffData.postFixOffset == 0) {//these 2 edge cases prevent the prefix or the postfix from being
+            // repeated when you append text to the beginning of the text or the end of the text
+            oldString.substring(0, diffData.prefixEnd) + diffData.diff
+        } else if (diffData.prefixEnd == 0) {
+            diffData.diff + oldString.substring(oldString.length - 1 - diffData.postFixOffset)
+        } else {
+            oldString.substring(0, diffData.prefixEnd) + diffData.diff +
+                    oldString.substring(oldString.length - 1 - diffData.postFixOffset)
+        }
+    }
+
     fun setValue(toBind: KVar<String>, updateOn: String = "input") {
         setValue(toBind as KVal<String>)
 
         // TODO: Would be really nice if it just did a diff on the value and sent that, rather than the
         //       entire value each time PARTICULARLY for large inputs
-        on(retrieveJs = "${jsExpression}.value").event(updateOn, Event::class) {
-            toBind.value = it.retrieved ?: error("No value was retrieved")
+        on(retrieveJs = "get_diff_changes(${jsExpression})").event(updateOn, Event::class) {
+            //I think this line could be made better to look at. But I think it works.
+            //I hate having to use the !!, it'd be better to catch the null and do some error handling
+            //But, I don't know really know what this function should do if it's sent a bad input
```

Comment:
It should probably throw an exception if the input is unexpected, similar to what was there before `it.retrieved ?: error("No value was retrieved")`